### ISSUE TITLE
Issue with Login Redirect and sessionStorage Error

### ISSUE THREAD

**Issue by sponnify** [StanGirard/quivr#844]
Description: When attempting to log in, the Quivr application encounters an error related to sessionStorage, which prevents the proper redirect after a successful login. This issue seems to be affecting users on Windows.

Steps to Reproduce:

1. Open the Quivr application on a machine running Windows 11.
2. Click on the "Login" button on the login page.
3. Enter valid login credentials (username and password).
4. Submit the login form by clicking on the "Submit" button.
5. Observe the error message in the Powershell console or docker console.

Error Message:
```
backend-core  | INFO:     Application startup complete.
web           | ReferenceError: sessionStorage is not defined
web           |     at redirectToLogin (/app/.next/server/app/upload/page.js:1053:5)
web           |     at UploadPage (/app/.next/server/app/upload/page.js:906:51)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:121:272)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:127:11)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
web           |     at Ke (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:131:155)
web           |     at Me (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:130:289)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:350)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:126:189)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
backend-core  | INFO:     172.18.0.1:33290 - "GET /brains/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.18.0.1:33302 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.18.0.1:33312 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
```

Note: The error occurs in the `useLogin.ts` file at line 35.

This issue is critical as it prevents users from accessing their accounts after a successful login. Any assistance in resolving this error would be greatly appreciated.

System Information:

- Operating System: Windows 11
- Docker Version: Docker version 24.0.2, build cb74dfc
- Docker Compose Version: Docker Compose version v2.19.1

Thank you for your attention to this matter! 🙏

**Comment by sponnify** [comment]
> Which browser are you using please @sponnify ?

Originally using Firefox, but this issue occurs in Edge as well.

**Comment by neumartin** [comment]
Same problem in Fedora Workstation 38 and the latest Chrome, Chromium, and Firefox.

```
web           | ReferenceError: sessionStorage is not defined
web           |     at redirectToLogin (/app/.next/server/app/upload/page.js:1093:5)
web           |     at UploadPage (/app/.next/server/app/upload/page.js:906:51)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:121:272)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:127:11)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
web           |     at Ke (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:131:155)
web           |     at Me (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:130:289)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:350)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:126:189)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
```

**Comment by rwilkin** [comment]
I can confirm this error comes up in Firefox, Edge and Brave. Does not happen with Chrome.
web           | ReferenceError: sessionStorage is not defined
web           |     at redirectToLogin (/app/.next/server/app/upload/page.js:1129:5)
web           |     at UploadPage (/app/.next/server/app/upload/page.js:971:51)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:121:272)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:127:11)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
web           |     at Ke (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:131:155)
web           |     at Me (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:130:289)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:350)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:126:189)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)

**Comment by CTGS-Innovations** [comment]
Check that your supabase project was not paused on supabase.. This was my issue and restoring it solved the "Session Storage issue"

**Comment by akkiverma** [comment]
```
web           | ReferenceError: sessionStorage is not defined
web           |     at redirectToLogin (/app/.next/server/app/upload/page.js:1129:5)
web           |     at UploadPage (/app/.next/server/app/upload/page.js:971:51)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:121:272)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:127:11)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
web           |     at Ke (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:131:155)
web           |     at Me (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:130:289)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:350)
web           |     at Je (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:126:189)
web           |     at Z (/app/node_modules/next/dist/compiled/react-dom/cjs/react-dom-server.edge.production.min.js:128:91)
backend-core  | INFO:     172.19.0.1:42324 - "OPTIONS /brains/ HTTP/1.1" 200 OK
backend-core  | INFO:     172.19.0.1:42328 - "OPTIONS /brains/default/ HTTP/1.1" 200 OK
backend-core  | INFO:     172.19.0.1:42342 - "OPTIONS /brains/default/ HTTP/1.1" 200 OK
backend-core  | INFO:     172.19.0.1:42328 - "GET /brains/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.19.0.1:42324 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.19.0.1:42342 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
```

Have same error, then I created a storage bucket at superbase, after that the error reduced to

```
backend-core  | INFO:     172.19.0.1:45146 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.19.0.1:45142 - "GET /brains/ HTTP/1.1" 403 Forbidden
backend-core  | INFO:     172.19.0.1:45146 - "GET /brains/default/ HTTP/1.1" 403 Forbidden
```

not sure what to do now. The project is not paused, did not saw any such thing there.

**Comment by akkiverma** [comment]
Console also shows this if that is helpful

`web           | react-i18next:: You will need to pass in an i18next instance by using initReactI18next
web           | i18next: languageChanged en
web           | i18next: initialized {
web           |   debug: true,
web           |   initImmediate: true,
web           |   ns: [ 'translation' ],
web           |   defaultNS: 'translation',
web           |   fallbackLng: [ 'en' ],
web           |   fallbackNS: false,
web           |   supportedLngs: false,
web           |   nonExplicitSupportedLngs: false,
web           |   load: 'all',
web           |   preload: false,
web           |   simplifyPluralSuffix: true,
web           |   keySeparator: '.',
web           |   nsSeparator: ':',
web           |   pluralSeparator: '_',
web           |   contextSeparator: '_',
web           |   partialBundledLanguages: false,
web           |   saveMissing: false,
web           |   updateMissing: false,
web           |   saveMissingTo: 'fallback',
web           |   saveMissingPlurals: true,
web           |   missingKeyHandler: false,
web           |   missingInterpolationHandler: false,
web           |   postProcess: false,
web           |   postProcessPassResolved: false,
web           |   returnNull: false,
web           |   returnEmptyString: true,
web           |   returnObjects: false,
web           |   joinArrays: false,
web           |   returnedObjectHandler: false,
web           |   parseMissingKeyHandler: false,
web           |   appendNamespaceToMissingKey: false,
web           |   appendNamespaceToCIMode: false,
web           |   overloadTranslationOptionHandler: [Function: handle],
web           |   interpolation: {
web           |     escapeValue: true,
web           |     format: [Function: bound format],
web           |     prefix: '{{',
web           |     suffix: '}}',
web           |     formatSeparator: ',',
web           |     unescapePrefix: '-',
web           |     nestingPrefix: '$t(',
web           |     nestingSuffix: ')',
web           |     nestingOptionsSeparator: ',',
web           |     maxReplaces: 1000,
web           |     skipOnVariables: true
web           |   },
web           |   lng: 'en',
web           |   resources: {
web           |     en: {
web           |       brain: [Object],
web           |       chat: [Object],
web           |       config: [Object],
web           |       explore: [Object],
web           |       login: [Object],
web           |       logout: [Object],
web           |       signUp: [Object],
web           |       translation: [Object],
web           |       updatePassword: [Object],
web           |       upload: [Object],
web           |       user: [Object],
web           |       delete_brain: [Object]
web           |     },
web           |     es: {
web           |       brain: [Object],
web           |       chat: [Object],
web           |       config: [Object],
web           |       explore: [Object],
web           |       login: [Object],
web           |       logout: [Object],
web           |       signUp: [Object],
web           |       translation: [Object],
web           |       updatePassword: [Object],
web           |       upload: [Object],
web           |       user: [Object],
web           |       delete_brain: [Object]
web           |     },
web           |     fr: {
web           |       brain: [Object],
web           |       chat: [Object],
web           |       config: [Object],
web           |       explore: [Object],
web           |       login: [Object],
web           |       logout: [Object],
web           |       signUp: [Object],
web           |       translation: [Object],
web           |       updatePassword: [Object],
web           |       upload: [Object],
web           |       user: [Object],
web           |       delete_brain: [Object]
web           |     }
web           |   },
web           |   ignoreJSONStructure: true
web           | }`

**Comment by mrshehzad1** [comment]
same issue on windows 11. anyone have solved this?

**Comment by ai-Ev1lC0rP** [comment]
I've run into this a few different times, and even gone down the path of creating a new deployment all together. It worked for a while, but after i had logged in from a different location (still on the same network) i started having the issue again. i think i'm going to try deploying the front and back-end separately if i can to see if that fixed is. If anyone else has any additional input or info on what steps you took to cause the issue, it would be helpful in troubleshooting.

**Comment by ai-Ev1lC0rP** [comment]
After more troubleshooting on my own, I realized the root cause was network config. I had been trying to access the app externally before properly configuring the networking permissions. Once I set up the rules to allow external access and tested locally via localhost first, it started working and stayed fixed.

I wanted to document what I did in case others hit this same problem. Setting up network access controls and validating locally before expanding access seems to be key. I also tried deploying front and back end separately, which helped isolate the issue.

If anyone else has run into this or found additional fixes, I'd be curious to hear what steps reproduced it for you. I was able to resolve it independently by locking down the networking and testing locally, but am open to other tips as I continue optimizing the configuration. Hopefully this helps anyone else troubleshooting similar issues!
